neqo-transport/src/server.rs

```diff
@@ -646,37 +630,35 @@ impl Server {
                 return Some(d);
             }
         }
-        qtrace!([self], "No packet to send still, run timers");
-        while let Some(c) = self.timers.take_next(now) {
-            if let Some(d) = self.process_connection(&c, None, now) {
+
+        qtrace!([self], "No packet to send still, check callbacks");
+        loop {
+            let connection = self
+                .connections
+                .borrow()
+                .values()
+                .find(|c| c.borrow().callback.map_or(false, |t| t <= now))
```

Comment:
Maybe wrap the ".callback.map_or()" bit in a function on the type itself.

Reply:
:+1: done in https://github.com/mozilla/neqo/pull/1784/commits/1a3b66412e890aaece900c1825952118dc6507c9.